app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/WalletConnectBridgeRepository.kt

```diff
@@ -179,9 +179,17 @@ class WalletConnectBridgeRepository @Inject constructor(
                                         TransactionExecutionRepository.Operation.CALL
                                     )
                                 }
-                                    .flatMap { (safe, tx) -> infoRepository.parseTransactionData(tx).map { txData -> safe to txData } }
+                                    .flatMap { (safe, tx) ->
+                                        infoRepository.checkRestrictedTransaction(safe, tx)
+                                            .flatMap(infoRepository::parseTransactionData)
+                                            .map { txData -> safe to txData }
+                                    }
                                     .subscribeBy(onError = { t ->
-                                        rejectRequest(id, 42, t.message ?: "Could not handle transaction").subscribe()
+                                        val message = when (t) {
+                                            is RestrictedTransactionException -> "This transaction is not allowed"
```

Comment:
should we add these messages to strings.xml?

Reply:
I left it out for now, since these are errors normally shown only on the developer console (in the dapp) but might make sense ... not sure :P